src/client/gl-base-technique.cc

```diff
@@ -227,6 +251,24 @@ GlBaseTechnique::ApplyUniformVariables(GLuint program_id, Camera* camera)
   }
 }
 
+void
+GlBaseTechnique::ApplyGlTexture()
+{
+  for (auto& texture : rendering_->gl_textures) {
+    if (auto gl_texture = texture.second.lock()) {
+      glBindTexture(texture.first, gl_texture->texture_id());
```

Comment:
We also have to specify the location of the texture to bind, like
`glActiveTexture(GL_TEXTURE0 + <location number>);`

And it's better to remove the element from the list, when the shared_ptr doesn't own a resource to match the bound texture in `ApplyGlTexture` and unbound texture in `UnapplyGlTexture`.